runtime/near-vm-logic/src/logic.rs

```diff
@@ -696,6 +696,23 @@ impl<'a> VMLogic<'a> {
         Ok(self.current_storage_usage)
     }
 
+    /// Returns the block hash for the block at the given height (if any). It can
+    /// only return a result for the most recent 256 blocks.
+    ///
+    /// # Cost
+    ///
+    /// `base + storage_read_base + storage_read_key_byte * 8 + write_register_base + write_register_byte * 32
```

Comment:
@nearmax @olonho @matklad 

I'm interested if you have any comments on the cost here.
Since the simple implementation I wrote for the backend uses the `Store` directly I figured the storage read costs would be appropriate.

Reply:
The reasoning here seems correct. Though, the specified formula is not exactly correct -- for the case where there isn't a hash, we don't pay `write_register_byte * 32` part, as we store zero-length vector.

Reply:
I don't think `storage_read_base` is necessarily the best base cost here -- Trie reads do not directly translate to database reads, but I think it is also fine to overestimate here.